src/psyclone/domain/lfric/kernel/common_arg.py

```diff
@@ -142,6 +148,69 @@ def check_fparser2(fparser2_tree, nargs=4, encoding=Fortran2003.Part_Ref):
                 f"{len(fparser2_tree.children[1].children)} in "
                 f"'{str(fparser2_tree)}'.")
 
+    @classmethod
+    def check_first_arg(cls, fparser2_tree, name, vector=False):
+        '''Check that the first metadata argument has the expected value.
+
+        :param fparser2_tree: the metadata encoded in an fparser2_tree
+        :type fparser2_tree: :py:class:`fparser.two.Fortran2003.Part_Ref` | \
+            :py:class:`fparser.two.Fortran2003.Structure_Constructor`
+        :param str name: the name used to describe this metadata.
+        :param bool vector: whether this is vector metadata.
+
+        :raises ValueError: if the first metadata argument has an \
+            incorrect value.
+
+        '''
+        idx = cls.form_arg_index
+        form = cls.get_arg(fparser2_tree, idx)
+        word = "as"
+        if vector:
+            form = form.split("*")[0].strip()
+            word = "in"
+        if not form.lower() == cls.form.lower():
+            raise ValueError(
+                f"{name}s should have {cls.form} {word} their first metadata "
+                f"argument, but found '{form}'.")
+
+    @classmethod
+    def check_remaining_args(cls, fparser2_tree, *args):
+        '''Check that the remaining untested metadata arguments have the
+        expected value. If they do not then re-raise the exception
+        from the class constructor, adding in positional information
+        and the metadata arguments to make it clearer where the
+        exception occured.
+
+        :param fparser2_tree: the metadata encoded in an fparser2_tree
+        :type fparser2_tree: :py:class:`fparser.two.Fortran2003.Part_Ref` | \
+            :py:class:`fparser.two.Fortran2003.Structure_Constructor`
+
```

Comment:
Please document the `args` argument. Possibly make the name a bit friendlier too - `metadata_args` perhaps?

Reply:
Added documentation, although it is unclear how to document *args! and changed name as suggested.

Reply:
Sergi uses "unwrapped dict" and I've just copied him :-)

Reply:
I've taken the liberty of Sergi-fying the docstring.